src/main/java/commands/AddCommand.java

```diff
@@ -0,0 +1,154 @@
+package commands;
+
+import java.time.LocalDateTime;
+
+import javafx.scene.layout.VBox;
+import parser.Parser;
+import storage.Storage;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+import tasklist.TaskList;
+import ui.Ui;
+
+public class AddCommand extends Command {
+
+	enum TaskType {
+		TODO, DEADLINE, EVENT
+	}
+
+	private Task task;
+	private TaskType taskType;
+
+	public AddCommand(String[] messageArray) {
+		super(messageArray);
+		setTaskType(messageArray[0]);
+	}
+
+	public TaskType getTaskType() {
+		return this.taskType;
+	}
+
+	/**
+	 * Set task type from user input.
+	 * 
+	 * @param message User input text.
+	 */
+	private void setTaskType(String message) {
+		String commandLower = message.toLowerCase();
+		this.taskType = commandLower.startsWith("todo")
+				? TaskType.TODO
+				: commandLower.startsWith("deadline")
+						? TaskType.DEADLINE
+						: TaskType.EVENT;
+	}
+
+	/**
+	 * Set task from task type.
+	 * 
+	 * @param inputArr        An array of strings from user text input.
+	 * @param ui              A service to render the page of GUI.
+	 * @param parser          A service to parse text input.
+	 * @param storage         A store that represents the data access object (DAO).
+	 * @param dialogContainer A container that holds all the rows of labels.
+	 * @param tasklist        All the tasks that are recorded.
+	 * @return boolean
+	 */
+	private boolean setTask(String[] inputArr, Ui ui, Parser parser, Storage storage, VBox dialogContainer,
+			TaskList tasklist) {
+		String description = parser.sliceArrAndConcate(inputArr, 1, inputArr.length);

```

Comment:
I like the naming of the parser function as it describes very clearly its purpose and the intended output!